### ISSUE TITLE
Questions about implementation

### ISSUE THREAD

**Issue by yjhong89** [jy0205/Pyramid-Flow#115]
Hi! Thanks for sharing great work!
Some questions about implementation

### What sample_size mean in PyramidDiffusionMMDiT
- https://github.com/jy0205/Pyramid-Flow/blob/fa4e1e52f71a46beb8bb29d1d3252232f3a47821/pyramid_dit/modeling_pyramid_mmdit.py#L66
- What role of this? It seems be fixed (as 128) and determine the base size of sincos spatial positional embedding.
- Is should be fixed even when fine-tuning to other resolution ?

### Temporal Causal Attention
- https://github.com/jy0205/Pyramid-Flow/blob/fa4e1e52f71a46beb8bb29d1d3252232f3a47821/pyramid_dit/modeling_pyramid_mmdit.py#L321
- It seems that temporal causal attention is not applied when using flash attention. Why??

THanks!

**Comment by yjhong89** [comment]
- Another question
### Spatial Pyramid jump point
- https://oss.navercorp.com/youngjun-hong/Pyramid-Flow/blob/60a9be3f5f240689e0cfe6f6b14725d5c1de0d94/pyramid_dit/pyramid_dit_for_video_gen_pipeline.py#L268 
- This is Equation 27 in paper ?? Then alpha and beta implementation seems not corresponding exactly to eq27. Would you give me something missed point for understanding ??
![image](https://github.com/user-attachments/assets/79712570-641d-49ca-8b0d-fb39e5859380)

**Comment by jy0205** [comment]
Thanks for your attention!
1. The `sample_size` is the setting of original SD3, it means the size of latent. This param is not used in our implementation. Just a inherit from sd3. It can be fixed during training.
2. The temporal causal is not applied when using flash attention. Because the flash attention does not support customized attention masks now.  We use block-wise causal attention, which means the token within a frame can not attend to its subsequent ones (not the traditional causal attention in LLM).
3. This is the implementation of Equation 27. You may misunderstand some details.

**Comment by yjhong89** [comment]
Thanks for answer!

- So temporal causal means that the token within a frame not attend to subsequent ones but attend to precedent ones, right ?

**Comment by feifeiobama** [comment]
> Thanks for answer!
> 
> * So temporal causal means that the token within a frame not attend to subsequent ones but attend to precedent ones, right ?

That is correct.

**Comment by yjhong89** [comment]
Thanks!
